### PR TITLE
Azure support of Active Directory credentinals

### PR DESCRIPTION
**Pull request by abraverm**

### CODE REVIEW

File: linchpin/provision/roles/azure/tasks/provisionVM.yml [hunk 1]
```diff
@@ -39,7 +42,7 @@
   poll: 0
   register: res_def_output
   when: _async
```
Comment (JunqiZhang0) [inline linchpin/provision/roles/azure/tasks/provisionVM.yml]: Should add this back?  not debug_mode

File: linchpin/provision/roles/azure/tasks/provisionVM.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ---
 - name: "Provisioning Azure VM when not async"
   azure_rm_virtualmachine:
+    ad_user: "{{ auth_var['user'] | default(omit) }}"
+    password: "{{ auth_var['password'] | default(omit) }}"
     client_id: "{{ auth_var['client_id'] | default(omit) }}"
```
Comment (JunqiZhang0) [inline linchpin/provision/roles/azure/tasks/provisionVM.yml]: Didn't delete old creds like (client_id, subscription_id), ansible will only detect one way of authentication

Comment (abraverm) [reply linchpin/provision/roles/azure/tasks/provisionVM.yml]: I left it on purpose for "IAM" method of authentication, checkout the doc.

File: linchpin/provision/roles/azure/tasks/provisionVM.yml (hunk 1 above)
Comment (abraverm) [reply linchpin/provision/roles/azure/tasks/provisionVM.yml]: you are right, I was lazy and forgot about it.

**Comment by abraverm** [comment]
[test]

**Comment by abraverm** [comment]
[test]

**Comment by abraverm** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by abraverm** [comment]
@samvarankashyap, is it possible to run tests with two

**Pull request closed without merging**